weston/weston-init

```diff
@@ -28,11 +28,72 @@ startup-animation=fade
 
 IEOF
 
+WPE_WESTON_OUTPUT_MAX_W=${WPE_WESTON_OUTPUT_MAX_W:-0}
+WPE_WESTON_OUTPUT_MAX_H=${WPE_WESTON_OUTPUT_MAX_H:-0}
+
+# Limit resolution if necessary
+if [ "$WPE_WESTON_OUTPUT_MAX_W" -gt 0 -o "$WPE_WESTON_OUTPUT_MAX_H" -gt 0 ]; then
+    for outp in /sys/class/drm/card*-*; do
+        # no outputs
+        if [ ! -d "$outp" ]; then
+            break
+        fi
+        # skip if not enabled
+        if [ "$(cat ${outp}/enabled)" = "disabled" ]; then
+            continue
+        fi
+        # get the output name
+        outn=${outp#*-}
+        # check the modes
+        moden=1
+        foundres=
+        for mode in $(cat "${outp}/modes"); do
+            wres=${mode%x*}
+            hres=${mode#*x}
+            # skip interlaced modes and other junk
+            if ! [ "$wres" -eq "$wres" -a "$hres" -eq "$hres" ] 2> /dev/null; then
+                moden=$(($moden + 1))
+                continue
+            fi
+            if [ "$WPE_WESTON_OUTPUT_MAX_W" -gt 0 ]; then
```

Comment:
This could be refactored into a function:

```sh
function is_out_of_max_range
{
    RES=$1
    MAX=$2
    [[ "$MAX" -gt 0 && "$RES" -gt "$MAX" ]] && return true
    return false
}

is_out_of_max_range "$wres" "$WPE_WESTON_OUTPUT_MAX_W" && moden=$(($moden + 1)) && continue
is_out_of_max_range "$hres" "$WPE_WESTON_OUTPUT_MAX_H" && moden=$(($moden + 1)) && continue

```

Reply:
yeah that's probably fine, but again your syntax is a bashism :)